src/internal_events/aws_kinesis_firehose.rs

```diff
@@ -51,11 +36,21 @@ impl<'a> InternalEvent for AwsKinesisFirehoseRequestError<'a> {
         error!(
             message = "Error occurred while handling request.",
             error = ?self.error,
+            error_type = "http_error",
+            code = %self.code,
+            stage = "receiving",
             internal_log_rate_secs = 10
         );
     }
 
     fn emit_metrics(&self) {
+        counter!(
+            "component_errors_total", 1,
+            "stage" => "receiving",
+            "error" => self.code.to_string(),
+            "error_type" => "http_error",
+        );
```

Comment:
This is a little confused. I think `code` should become `error_code` or `status_code`, but then what to do about the `error` property here? It's a free-format string, which could lead to issues when output as a tag, but it is the proper value to go into the metric. Thoughts?

Reply:
I replaced the `error` value by the status code `reason-phrase` so that the error makes sense ("Not Found", "Internal Server Error", "I'm a teapot").

Reply:
And I also changed `code` to `error_code`